x-pack/legacy/plugins/siem/public/components/timeline/body/column_headers/header/index.tsx

```diff
@@ -96,9 +96,79 @@ interface Props {
 }
 
 /** Renders a header */
-export class Header extends React.PureComponent<Props> {
-  public render() {
-    const { header } = this.props;
+export const Header = React.memo<Props>(
+  ({
+    header,
+    onColumnRemoved,
+    onColumnResized,
+    onColumnSorted,
+    onFilterChange = noop,
+    setIsResizing,
+    sort,
+  }) => {
+    const onClick = () => {
+      if (header.aggregatable) {
+        onColumnSorted!({
+          columnId: header.id,
+          sortDirection: getNewSortDirectionOnClick({
+            clickedHeader: header,
+            currentSort: sort,
+          }),
+        });
+      }
+    };
+
+    const onResize: OnResize = ({ delta, id }) => {
+      onColumnResized({ columnId: id, delta });
+    };
+
+    const renderActions = (isResizing: boolean) => {
+      setIsResizing(isResizing);
```

Comment:
I think we should investigate this `setIsResising` here because I am sure it is adding some side effect.

Reply:
lets pair on this